### DISCUSSION TITLE
Csla 6 fluent configuration - should the AddServerSideDataPortal() method be renamed?

### DISCUSSION THREAD

**Discussion by TheCakeMonster** [MarimerLLC/csla#2694] [Ideas]
In order to fix some bugs in the fluent configuration system for Csla 6 I have submitted a change in PR #2686 to what the AddServerSideDataPortal method does. If this change is accepted the method will no longer add anything itself; instead, it's a method through which configuration can be defined. A separate, internal method does the actual application of the configuration - the additions.

I can see arguments for a few different names:

1. Retaining the existing method name with its Add prefix because this is a convention for methods associated with configuring DI.
2. ServerSideDataPortal or ServerSide because these are slightly more fluent in explaining what we are configuring.
3. ConfigureServerSide because this is more fluent in explaining what we are doing.

Options 2 and 3 would be a breaking change for lots of people, as I think this method has existed since at least 2018 and will have been live in multiple versions. As such, I'm not sure it is worth the pain of changing it, but I thought I would raise it for discussion since I'm here talking about config.

**Comment by rockfordlhotka** [comment]
If we're going to break things, now is the time. To be fair, the CSLA 6 fluent API _already_ is a serious breaking change, so this seems like a relatively minor thing to add to the list of existing changes.

Also, I'd argue that these changes affect only the startup code - they aren't replicated throughout anyone's codebase (except perhaps for unit testing).

**Reply by swegele** [reply]
My thoughts exactly when I read (and agreed with) the recommendations in this discussion...I thought "Well now's the time..."
